producer-kafka/src/main/java/io/stargate/producer/kafka/KafkaCDCProducer.java

```diff
@@ -61,25 +64,42 @@ public KafkaCDCProducer(MetricRegistry registry, ConfigStore configStore) {
 
   /**
    * It registers the provided MetricRegistry in the Dropwizard shared metrics registry (see {@link
-   * SharedMetricRegistries}). The {@link DropwizardMetricsReporter} is getting the metrics registry
-   * via {@code SharedMetricRegistries.getOrCreate("default")} in the constructor.
+   * SharedMetricRegistries}). The {@link io.dropwizard.kafka.metrics.DropwizardMetricsReporter} is
+   * getting the metrics registry via {@code SharedMetricRegistries.getOrCreate("default")} in the
+   * constructor.
    */
   private void registerMetrics(MetricRegistry registry) {
     SharedMetricRegistries.add("default", registry);
   }
 
   @Override
   public CompletableFuture<Void> init() {
+    LOGGER.debug("Initializing KafkaCCDProducer");
     CDCKafkaConfig cdcKafkaConfig = configLoader.loadConfig(configStore);
-
+    LOGGER.info("Using config: {}", cdcKafkaConfig);
     this.mappingService = new DefaultMappingService(cdcKafkaConfig.getTopicPrefixName());
     this.schemaProvider =
         new SchemaRegistryProvider(cdcKafkaConfig.getSchemaRegistryUrl(), mappingService);
     this.keyValueConstructor = new KeyValueConstructor(schemaProvider);
 
     kafkaProducer =
         CompletableFuture.supplyAsync(
-            () -> new CompletableKafkaProducer<>(cdcKafkaConfig.getKafkaProducerSettings()));
+            () -> {
+              Thread currentThread = Thread.currentThread();
+              ClassLoader ldr = currentThread.getContextClassLoader();
+              try {
+                // Note: Apache Kafka expects to find certain dependencies
+                // using the thread context class loader.
+                // However, in our OSGi framework,
+                // the default context class loader is the system class loader,
+                // which does not contain module jars.
+                currentThread.setContextClassLoader(contextClassLoader);
+                return new CompletableKafkaProducer<>(cdcKafkaConfig.getKafkaProducerSettings());
+              } finally {
+                currentThread.setContextClassLoader(ldr);
+              }
+            },
+            Executors.newSingleThreadExecutor());
```

Comment:
The executor's thread will keep running after the task is done, will it not?

Reply:
You are right, it will not stop the executor and I think we shouldn't stop it. The reason for this is that we don't want to block here and let the caller decide.

Reply:
I've did add shutdown() in the close().